getting errors when i generate a route

i generated a route using using 
ng generate route ROUTE

if i open the app i get lots of errors like this 


browser_adapter.js:77 Error: DI Exception
    at NoProviderError.BaseException [as constructor] (exceptions.js:17)
    at NoProviderError.AbstractProviderError [as constructor] (reflective_exceptions.js:39)
    at new NoProviderError (reflective_exceptions.js:75)
    at ReflectiveInjector_._throwOrNull (reflective_injector.js:776)
    at ReflectiveInjector_._getByKeyDefault (reflective_injector.js:804)
    at ReflectiveInjector_._getByKey (reflective_injector.js:767)
    at ReflectiveInjector_.get (reflective_injector.js:576)
    at ElementInjector.get (element_injector.js:23)
    at DebugAppView._View_DemocracyAppComponent0.createInternal (DemocracyAppComponent.template.js:24)
    at DebugAppView.AppView.create (view.js:66)

--- a/addon/ng2/blueprints/route/index.js
+++ b/addon/ng2/blueprints/route/index.js
@@ -214,7 +214,11 @@ module.exports = {
     }
 
     let isAppComponent = false;
-    if (parentFile == path.join(this.dynamicPath.dir, this.project.name() + '.component.ts')) {
+    let appComponentFile = 
+      path.join(this.project.root, 
+        this.dynamicPath.dir, 
+        this.project.name() + '.component.ts');
+    if (parentFile == appComponentFile) {
       isAppComponent = true;
     }
 
@@ -269,6 +273,7 @@ module.exports = {
     // Add the provider, only on the APP itself.
     if (isAppComponent) {
       content = _insertImport(content, 'ROUTER_DIRECTIVES', '@angular/router');
+      content = _insertImport(content, 'ROUTER_PROVIDERS', '@angular/router');
       content = content.replace(/(@Component\(\{)([\s\S\n]*?)(\n\}\))/m, function (_, prefix, json, suffix) {
         const m = json.match(/(^\s+providers:\s*\[)([\s\S\n]*)(\]\s*,?.*$)/m);
         if (m) {

